fenix/app/src/androidTest/java/org/mozilla/fenix/ui/SettingsAddonsTest.kt

```diff
@@ -116,24 +114,37 @@ class SettingsAddonsTest {
     }
 
     // TestRail link: https://testrail.stage.mozaws.net/index.php?/cases/view/561600
-    // Installs uBlock add-on and checks that the app doesn't crash while loading pages with trackers
+    // Installs 3 add-on and checks that the app doesn't crash while navigating the app
     @SmokeTest
     @Test
     fun noCrashWithAddonInstalledTest() {
         // setting ETP to Strict mode to test it works with add-ons
         activityTestRule.activity.settings().setStrictETP()
 
-        val addonName = "uBlock Origin"
+        val uBlockAddon = "uBlock Origin"
+        val darkReaderAddon = "Dark Reader"
+        val privacyBadgerAddon = "Privacy Badger"
         val trackingProtectionPage = getEnhancedTrackingProtectionAsset(mockWebServer)
 
         addonsMenu {
-            installAddon(addonName)
-            verifyAddonInstallCompleted(addonName, activityTestRule)
+            installAddon(uBlockAddon)
+            verifyAddonInstallCompleted(uBlockAddon, activityTestRule)
+            closeAddonInstallCompletePrompt()
+            installAddon(darkReaderAddon)
+            verifyAddonInstallCompleted(darkReaderAddon, activityTestRule)
+            closeAddonInstallCompletePrompt()
+            installAddon(privacyBadgerAddon)
+            verifyAddonInstallCompleted(privacyBadgerAddon, activityTestRule)
```

Comment:
I'm a little worried for potential network issues here as we're widening the surface.

Looking at 

https://github.com/mozilla-mobile/firefox-android/blob/7fad484fd429a1696de812ab09428f3fe9ab7c49/fenix/app/src/androidTest/java/org/mozilla/fenix/ui/robots/SettingsSubMenuAddonsManagerRobot.kt#L86 - can we add some safety to the install helper? Or does that happen below in `verifyAddonInstallCompleted` ?

Reply:
As discussed with Aaron, I added the verifyAddonInstallCompleted method inside the installAddon() helper to make sure we catch an error while installing the add-on and retry.